syncd/FlexCounter.cpp

```diff
@@ -2122,6 +2233,19 @@ void FlexCounter::addCounter(
 
             setSwitchDebugCounterList(vid, rid, switchCounterIds);
         }
+        else if (objectType == SAI_OBJECT_TYPE_MACSEC_SA && field == MACSEC_SA_ATTR_ID_LIST)
+        {
+            std::vector<sai_macsec_sa_attr_t> macsecSAIds;
+
+            for (const auto &str : idStrings)
+            {
+                sai_macsec_sa_attr_t attr;
+                sai_deserialize_macsec_sa_attr(str.c_str(), attr);
```

Comment:
no need for c_str() since parameter is string already

Reply:
Done, please review it.